app/src/main/java/com/example/diceroller/MainActivity.kt

```diff
@@ -4,122 +4,80 @@ import android.os.Bundle
 
-    private var _binding: ActivityMainBinding? = null
-    private val binding get() = _binding!!
-
-    private lateinit var viewModel: RolledDiceViewModel
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var viewModel: DiceRollerViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        _binding = ActivityMainBinding.inflate(layoutInflater)
-        val view = binding.root
-        setContentView(view)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
         supportActionBar?.hide()
 
-        viewModel = ViewModelProvider(this).get(RolledDiceViewModel::class.java)
+        viewModel = ViewModelProvider(this)[DiceRollerViewModel::class.java]
 
         setListeners()
+        observeState()
     }
 
-    private fun setListeners() {
-        val navigationMenu = binding.navView
-
-        val menuButton = binding.menuButton
-
-        toggleNavigation(menuButton, navigationMenu)
-
-        navigationMenu.setNavigationItemSelectedListener(this)
-    }
-
-
-    private fun toggleNavigation(menuButton: View, navigationMenu: View) {
-        menuButton.setOnClickListener {
-            when (navigationMenu.visibility) {
-                View.GONE -> {
-                    navigationMenu.visibility = View.VISIBLE
-                }
-                View.VISIBLE -> {
-                    navigationMenu.visibility = View.GONE
-                }
-                else -> {
-                    navigationMenu.visibility = View.GONE
+    private fun observeState() {
+        lifecycleScope.launchWhenStarted {
+            viewModel.viewState.collect {
+                if (it is ViewState.RollDie) {
+                    binding.title.text =
+                        getString(R.string.toolbar_label, it.diceType)
                 }
             }
         }
     }
 
-    override fun onNavigationItemSelected(selectedItem: MenuItem): Boolean {
-        val navigationMenu = binding.navView
-        val fragmentContainer = binding.fragmentContainerView.id
-        val diceFragment = R.id.diceFragment
-        val listOfDiceFragment = R.id.rolledDicesListFragment
-        when(selectedItem.itemId) {
-            R.id.D4DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD4()
-            }
-
-            R.id.D6DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD6()
-            }
-
-            R.id.D8DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD8()
-            }
-
-            R.id.D10DiceMenuItem -> {
-                navigateToFragment(fragmentContainer, diceFragment)
-                viewModel.triggerD10()
-            }
```

Comment:
‼️ I can tell that you did this as a way to update the state of the Fragment. High praise for finding a way to create the behavior you wanted. 

This isn't ideal though, thing of a Fragment like a window on your desktop. (In reality its nothing like that but for this explanation its close enough). Would it make sense to close the window and reopen it because you want to show different things on screen? Wouldn't you prefer use to update the assets on screen? (change the image, change the text, etc).

Take a look at the changes I made. I challenge you to explain to me what's different.